meerk40t/core/bindalias.py

```diff
@@ -135,105 +232,58 @@ def save_keymap_alias(self):
         keys.clear_persistent()
         alias.clear_persistent()
 
-        for key in self.keymap:
+        for key, value in self.context.keymap.items():
             if key is None or len(key) == 0:
                 continue
-            keys.write_persistent(key, self.keymap[key])
+            keys.write_persistent(key, value)
 
-        for key in self.alias:
+        for key, value in self.context.alias.items():
             if key is None or len(key) == 0:
                 continue
-            alias.write_persistent(key, self.alias[key])
+            alias.write_persistent(key, value)
 
     def boot_keymap(self):
-        self.keymap.clear()
-        prefs = self.context.derive("keymap")
-        prefs.kernel.load_persistent_string_dict(prefs.path, self.keymap, suffix=True)
-        if not len(self.keymap):
+        context = self.context
+        context.keymap.clear()
+        prefs = context.derive("keymap")
+        prefs.kernel.load_persistent_string_dict(prefs.path, context.keymap, suffix=True)
+        if not len(context.keymap):
             self.default_keymap()
+            return
+        for key, values in DEFAULT_KEYMAP.items():
+            if key in context.keymap and context.keymap[key] in values[1:]:
```

Comment:
Tested and this was insufficient.

I added a key `control+q` equal to `testing`
```python
     "control+q": ("testing",),
```
And ran this code. As control+q is not in the context.keymap it did not add my new key. I'll go ahead and accept since this implies *update* but I think maybe a third thing, where unless they bound that key for something else or specifically set it to none, it should rebind that key. Part of the problem is that new binds also do not get added.

I do not know how you tell a new bind from a bind that previously existed and was then set to none, but it seems like the spirit of the changes should address this problem. If not, say the word and I'll approve this.

Reply:
If I understand this correctly, you added a DEFAULT_KEYMAP entry for an entirely new key not currently mapped in your settings in the code and it was NOT added to the bindings?

If this is what you are reporting, I agree - and 2 lines of additional code are required (plus two for the similar alias code).